lib/mocha/mocha.ts

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+((context) => {
+  var Mocha = context['Mocha'];
+  
+  if (typeof Mocha === 'undefined') {
+    throw new Error('Missing Mocha.js');
+  }
+
+  if (typeof Zone === 'undefined') {
+    throw new Error('Missing Zone.js');
+  }
+
+  let ProxyZoneSpec = Zone['ProxyZoneSpec'];
+
+  if (!ProxyZoneSpec) {
+    throw new Error('Missing ProxyZoneSpec');
+  }
+
+  if(Mocha['__zone_patch__']){
+    throw new Error('"Mocha" has already been patched with "Zone".');
+  }
+
+  Mocha['__zone_patch__'] = true;
+
+  let rootZone = Zone.current;
+
+  function wrapCallbackInProxyZone(fn: Function): Function {
```

Comment:
I did some more updates on this.
The reason is that I noticed that Mochas **after/before**, **suiteSetup/suiteTeardown** was wrapped false.
**after/before**, **suiteSetup/suiteTeardown** is triggered after all tests in a suite is completed, which means that it should have it own zone, like describe.
Now "describe" has it own zone (syncZone), each "suite", the context for that is after/before/suiteSetup/suiteTeardown, (suiteZone) and each test (testZone).
I have added test to verify this.